iree/compiler/Translation/IREEVM.cpp

```diff
@@ -103,10 +106,44 @@ static LogicalResult translateFromMLIRToVMBytecodeModuleWithFlags(
                                              bytecodeTargetOptions, output);
 }
 
+#ifdef IREE_HAVE_EMITC_DIALECT
+LogicalResult translateFromMLIRToVMCModule(
+    ModuleOp moduleOp, IREE::HAL::TargetOptions executableOptions,
+    llvm::raw_ostream &output) {
+  // Convert from our source to a vm.module in canonical form.
+  // After this completes we have a non-bytecode-specific vm.module that we
+  // could lower to other forms (LLVM IR, C, etc).
+  PassManager passManager(moduleOp.getContext());
+  mlir::applyPassManagerCLOptions(passManager);
+  IREE::Flow::buildFlowTransformPassPipeline(passManager);
+  IREE::HAL::buildHALTransformPassPipeline(passManager, executableOptions);
+  IREE::VM::buildVMTransformPassPipeline(passManager);
+  passManager.addPass(mlir::iree_compiler::IREE::createDropCompilerHintsPass());
+  if (failed(passManager.run(moduleOp))) {
+    return moduleOp.emitError() << "conversion from source -> vm failed";
+  }
```

Comment:
Shared function with the above?

Reply:
I moved the common code into a function. I'm not sure about the style guide though.

Reply:
@GMNGeoffrey I think the question is whether the function should be forward declared or not. However, a forward declaration is not necessary as long as the order of the functions is unchanged. I think it should not be declared in the header, because we don't want to expose it to the outside. Is there any rule for this?

Reply:
Yeah I think no need for a forward declaration. Since this is private to the cpp file, I would use internal linkage, and mark the helper function `static`

https://google.github.io/styleguide/cppguide.html#Unnamed_Namespaces_and_Static_Variables
https://llvm.org/docs/CodingStandards.html#anonymous-namespaces

Reply:
Okay, I made the function static and removed the comment.